styx-common/src/main/java/com/spotify/styx/util/WorkflowValidator.java

```diff
@@ -133,6 +138,12 @@ public static Builder newBuilder(DockerImageValidator dockerImageValidator) {
       }
     });
 
+    cfg.secret().ifPresent(secret -> {
+      if (secretWhitelist != null && !secretWhitelist.contains(secret.name())) {
```

Comment:
I would rather initialize `secretWhitelist` to `Set.of()` (the empty set) in the `Builder` class so we can simplify this line to be just 

```java
if (!secretWhitelist.contains(secret.name())) {
```

Reply:
Unfortunately this validator is used by CLIMain.java where we can not apply any whitelist check. So `null` means skipping whitelist check.

Reply:
Unless we say `Set.of("*")` is a wildcard.

Reply:
Wildcard will make it more messy. As we discussed earlier the problem with the nulls are that we are using the same class in the cli and the service. What we need is actually split this class and make the validators composable but that is for another PR